File: Frontend/js/sensor.js
class Sensor {
    constructor(id, name, parameter) {
      this.id = id;
      this.name = name;
      this.parameter = parameter;
      this.value = 0;
      this.minValue = 0;
      this.maxValue = 100;
    }
  
    setValue(value) {
      this.value = value;
      if (this.value < this.minValue || this.value > this.maxValue) {
        this.sendAlert();
      }
    }
  
    sendAlert() {
      const message = `Sensor ${this.name} com valor ${this.value} fora do limite`;
      console.log(message);
      // Enviar mensagem para o Broker
      fetch('/api/alert', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        },
        body: JSON.stringify({
          topic: this.parameter,
          message: message
        })
      });
    }
  }
  
  const sensors = [
    new Sensor(1, "Temperatura", "temperatura"),
    new Sensor(2, "Umidade", "umidade"),
    new Sensor(3, "Velocidade", "velocidade")
  ];
  
  const sensorList = document.getElementById("sensors");
  sensors.forEach(sensor => {
    const li = document.createElement("li");
    li.textContent = `${sensor.name} - ${sensor.parameter}`;
    sensorList.appendChild(li);
  });

File: Backend/sensor.py
import paho.mqtt.client as mqtt

class Sensor:
  def __init__(self, id, name, parameter):
    self.id = id
    self.name = name
    self.parameter = parameter
    self.value = 0
    self.minValue = 0
    self.maxValue = 100
    self.client = mqtt.Client()
    self.client.connect("localhost", 1883)

  def sendAlert(self):
    message = f"Sensor {self.name} com valor {self.value} fora do limite"
    self.client.publish(f"{self.parameter}/alert", message)

sensors = [
  Sensor(1, "Temperatura", "temperatura"),
  Sensor(2, "Umidade", "umidade"),
  Sensor(3, "Velocidade", "velocidade")
]